### PR TITLE
Add date cutoff parameter

### LINKED ISSUE #136: Add support for relative date scrape cutoffs
Right now the CNN scraper goes back in time until there is nothing left to scrape.  This isn't actually useful, since we don't care about old claims (and in fact likely don't want to store them at all as we are discussing in #37).

We need to be able to configure the system to only look back a certain number of days when scraping.  For now I think we can have this be universal across all statement scrapers (as opposed to configuring each scraper cutoff separately).  It should be a relative date (e.g. "past 5 days") so that as time marches on the scraper stops scraping things from the past even if we wipe scrape longs, etc.

### PR DESCRIPTION
**Pull request by slifty**
This creates a new environment variable SCRAPE_DAY_HORIZON which specifies the number of days prior to the current date that should be scraped for claims.

It also adds a few tests and utility methods that can be used in other scraper pipelines eventually.

This continues the whole "call a URL a path" thing from #129 in the new `extractPublicationDateFromUrl` method.  We'll need to fix it here as well when we fix that issue.

Right now there is no way to say "scrape all known claims" since there's no real use case for that.

Resolves #136

### CODE REVIEW

File: src/server/models/index.js [hunk 1]
```diff
@@ -10,7 +11,13 @@ const env = config.NODE_ENV || ENV_NAMES.DEVELOPMENT
 const envConfig = sequelizeConfig[env]
 
 const db = {}
-const sequelize = new Sequelize(envConfig.url, envConfig)
+const sequelize = new Sequelize(
+  envConfig.url,
+  Object.assign(
+    { logging: msg => logger.debug(msg) },
```
Comment (reefdog) [inline src/server/models/index.js]: :squint: Why aren't we doing this in `sequelize-config.js`?

File: src/server/utils/scraper.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Disabling because we intend to have more exports in the future.
+/* eslint-disable import/prefer-default-export */
+import dayjs from 'dayjs'
+import config from '../config'
+
+export const isDateBeforeScrapeHorizon = date => dayjs()
```
Comment (reefdog) [inline src/server/utils/scraper.js]: My literary bent wants this to be called `isDateBeyondScrapeHorizon` for metaphorical consistency but this is fine and correct.

In any case does the `date` param have to be a DayJS object, or can it be a DayJS-parseable string? Worth documenting this function to make this clear.

File: src/server/queues/cnnTranscriptListCrawlerQueue/cnnTranscriptListCrawlerJobProcessor.js [hunk 3]
```diff
@@ -10,9 +12,11 @@ const scrapeTranscriptUrl = url => statementScraperQueue.add({ url })
 const processTranscriptUrl = async (url) => {
   const scraper = new CnnTranscriptStatementScraper(url)
   const recentScrapeTime = await scraper.getMostRecentScrapeTime()
-
+  const urlPublicationDate = extractPublicationDateFromTranscriptUrl(url)
```
Comment (reefdog) [inline src/server/queues/cnnTranscriptListCrawlerQueue/cnnTranscriptListCrawlerJobProcessor.js]: A neat shortcut! So much faster than having to actually scrape the transcript and parse out its date there. Hope they never stop using date-embedded URLs…

File: src/server/utils/__test__/cnn.test.js [hunk 4]
```diff
@@ -68,6 +69,24 @@ describe('utils/cnn', () => {
...
+    it('Should extract dates', () => {
+      const date = extractPublicationDateFromTranscriptUrl('/TRANSCRIPTS/1906/01/cnr.20.html')
+      expect(date)
+        .not.toBeNull()
+      expect(date.date())
+        .toEqual(1)
+      expect(date.year())
+        .toEqual(2019)
+      expect(date.month())
+        .toEqual(5)
```
Comment (reefdog) [inline src/server/utils/__test__/cnn.test.js]: This is all fine, but L77-82 could be shortened to `expect(date.format('YYYY-MM-DD').toBe('2019-06-01')`.

File: src/server/utils/__test__/scraper.test.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('isDateBeforeScrapeHorizon', () => {
+  const day = dayjs()
+  it('Should recognize dates before the date horizon', () => {
+    const earlyDate = day.subtract(config.SCRAPE_DAY_HORIZON + 1, 'day')
+    expect(isDateBeforeScrapeHorizon(earlyDate))
+      .toBe(true)
+  })
+  it('Should recognize dates after the date horizon', () => {
+    const earlyDate = day.add(1, 'day')
```
Comment (reefdog) [inline src/server/utils/__test__/scraper.test.js]: Rename `earlyDate` to `validDate` or something, yeah?

File: src/server/utils/cnn.js [hunk 6]
```diff
@@ -30,6 +35,25 @@ export const getFullCnnUrl = (url) => {
   return `http://cnn.com/${url}`
 }
 
+/**
+ * Pull out the publication date from a passed transcript url.
```
Comment (reefdog) [inline src/server/utils/cnn.js]: `url` → `URL`

File: src/server/utils/cnn.js [hunk 7]
```diff
@@ -30,6 +35,25 @@ export const getFullCnnUrl = (url) => {
...
+ *
+ * If the url is not a valid transcript URL, throw an error.
```
Comment (reefdog) [inline src/server/utils/cnn.js]: `If the url` → `If the URL`

File: src/server/models/index.js (hunk 1 above)
Comment (slifty) [reply src/server/models/index.js]: That isn't babel'd so we can't import the logger there.

Comment (reefdog) [reply src/server/models/index.js]: 👀 ah ok.

File: src/server/utils/cnn.js (hunk 6 above)
Comment (reefdog) [reply src/server/utils/cnn.js]: Actually gonna do this as a hotfix to master to grab the rest.

File: src/server/utils/cnn.js (hunk 7 above)
Comment (reefdog) [reply src/server/utils/cnn.js]: Actually gonna do this as a hotfix to master to grab the rest.

**Pull request merged**